### ISSUE TITLE
Replicants will cause errors when they contain a Date object

### ISSUE THREAD

**Issue by Lange** [nodecg/nodecg#369]
Originally reported by @Azenet in Gitter.

Setting a Replicant's value to be a `Date` object (or any property of the Replicant's value -- the `Date` object doesn't have to be the root) will cause errors when attempting to serialize that value for persistence or transmission across the wire.

At this time, I don't fully understand what the issue is. My guess is that something in the Proxyification process ([lib/replicant/shared.js#L288](https://github.com/nodecg/nodecg/blob/master/lib/replicant/shared.js#L288)) is breaking Date objects.

**Comment by Hoishin** [comment]
This would help investigation.
```
UNCAUGHT EXCEPTION! NodeCG will now exit.
TypeError: Method Date.prototype.valueOf called on incompatible receiver [object Object]
    at Proxy.toJSON (native)
    at Object.stringify (native)
    at process.nextTick.error (U:\nodecg\nodecg\lib\replicator.js:265:17)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickDomainCallback (internal/process/next_tick.js:122:9)
```

**Comment by Azenet** [comment]
@Hoishin this was when using outdated Node v6.9.4. Using Node v8.9.4:
```
UNCAUGHT EXCEPTION! NodeCG will now exit.
TypeError: this is not a Date object.
    at Proxy.valueOf (<anonymous>)
    at Proxy.[Symbol.toPrimitive] (<anonymous>)
    at Proxy.toJSON (<anonymous>)
    at JSON.stringify (<anonymous>)
    at process.nextTick (U:\nodecg\nodecg\lib\replicator.js:265:17)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickDomainCallback (internal/process/next_tick.js:218:9)
```

**Comment by Seldszar** [comment]
A potential fix could be to check if the value is actually a plain object here:

https://github.com/nodecg/nodecg/blob/505308036afa8556ba874b72731e124c93d161ef/lib/replicant/shared.js#L289

Like that, complex objects (like Date, RegExp, Buffer, classes, etc.) are ignored by the proxyfication process. I don't know the potential regressions but it as far I know, a replicant is only able to use types JSON can handle.

**Comment by Lange** [comment]
Closed by #370.
